### PR TITLE
Fix #1134 - Correct hourly forecast time offset

### LINKED ISSUE #1134: Hourly Forecast Timing Mismatch
### Steps to reproduce

Open Breezy Weather and navigate to the hourly forecast section.

### Expected behavior

The hourly forecast should align with the correct hours.

### Actual behavior

The forecast times are one hour earlier than they should be.

### Weather source used

China

### Breezy Weather version

5.2.4

### Android version

Android 13

### Device

Xiaomi MIX4

### Other details

Hello,

I have noticed an issue with the hourly forecast in Breezy Weather. When using "中国 (China)" as the weather source, the hourly predictions appear to be shifted by one hour earlier than they should be.

I have verified this discrepancy by comparing the hourly forecast with Xiaomi Weather, Huawei Weather, and OPPO Weather, which all show consistent and accurate timing that is one hour later than Breezy Weather.

## Example:

For instance, the forecast from 14:00 to 18:00 should be 32, 32, 31, 31, 30. However, Breezy Weather shows these values from 13:00 to 17:00.

The screenshot below shows this situation.

Screenshot_20240614-130740_Breezy Weather

| APP            | Now (1 PM) | 2 PM | 3 PM | 4 PM | 5 PM | 6 PM |
| -------------- | ---------- | ---- | ---- | ---- | ---- | ---- |
| Breezy Weather | 32℃        | 32℃  | 31℃  | 31℃  | 30℃  | 29℃  |
| Xiaomi Weather | 33℃        | 32℃  | 32℃  | 31℃  | 31℃  | 30℃  |
| Huawei Weather | /          | 32℃  | 32℃  | 31℃  | 31℃  | 30℃  |
| OPPO Weather   | 33℃        | 32℃  | 32℃  | 31℃  | 31℃  | 30℃  |

Thank you for your attention to this issue. I appreciate the hard work the team puts into this project and hope this can be resolved soon.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [X] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [X] I have updated the app to **[version 5.2.4 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)** and I can reproduce the issue in this version.
- [ ] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

### PR DESCRIPTION
**Pull request by JiunnTarn**

### CODE REVIEW

**Comment by papjul** [comment]
That's a workaround that may not work all the time. I'm under the impression that since you encounter issues, it means it's not based on `publishDate`? The lack of documentation of this source doesn't help :/

**Comment by JiunnTarn** [comment]
I understand your concerns, but since there are no official documents for this weather source and it might have been obtained through reverse engineering, we can only make inferences from the results.

image

From the results, it appears that Xiaomi Weather, Huawei Weather, and OPPO Weather all consider the first value of `forcastHoutly.temperature.value` to be the temperature for the next hour after `pubTime`. Since these three APP all interpret it this way, I believe this explanation is likely correct.

Additionally, judging from the weather source's domain name `weatherapi.market.xiaomi.com`, it belongs to `xiaomi.com`. Therefore, I trust that Xiaomi Weather's interpretation of the data returned by this API should be accurate.

In conclusion, I believe that the error lies in Breezy Weather's interpretation.

**Comment by papjul** [comment]
I didn't say I had the wrong interpretation, we definitely have an issue. But the question is that I base myself on pubTime but it might be something else instead

**Comment by JiunnTarn** [comment]
I understand your point. In this new commit, I use `forecastHourly.temperature.pubTime` as the base time for the hourly forecast (previously it was current's `pubTime`).

File: app/src/src_nonfreenet/org/breezyweather/sources/china/ChinaResultConverter.kt [hunk 1]
```diff
@@ -199,15 +198,16 @@ private fun getPrecipitationProbability(forecast: ChinaForecastDaily, index: Int
 }
 
 private fun getHourlyList(
-    publishDate: Date,
     location: Location,
     hourlyForecast: ChinaForecastHourly
 ): List<HourlyWrapper> {
     if (hourlyForecast.weather == null || hourlyForecast.weather.value.isNullOrEmpty()) return emptyList()
 
+    val hourlyListPubTime = hourlyForecast.temperature?.pubTime ?: return emptyList()
```
Comment (papjul) [inline app/src/src_nonfreenet/org/breezyweather/sources/china/ChinaResultConverter.kt:206]: I don't agree to return no hourly forecast if there is no `pubTime`. Either make it mandatory (not `null`) in the parsing, or make it fallback to current hour.

Comment (JiunnTarn) [reply app/src/src_nonfreenet/org/breezyweather/sources/china/ChinaResultConverter.kt]: Actually I refer to is this code in `getMinutelyList`.

https://github.com/breezy-weather/breezy-weather/blob/207bc1374e8f25d6a1099c153994826cfb39a504/app/src/src_nonfreenet/org/breezyweather/sources/china/ChinaResultConverter.kt#L241

But I agree with what you said. In this new commit, if `pubTime` is null, it will fallback to the current hour.

**Comment by papjul** [comment]
Thanks, could you make a single commit for easier maintenance of the Git history, please?

Instructions in [Contribution guide](https://github.com/breezy-weather/breezy-weather/blob/main/CONTRIBUTE.md#submit) if you're not familiar with Git.

**Comment by JiunnTarn** [comment]
Thanks, I merged my commits, hope I did it right.
